### PR TITLE
Danielgomeza1406/cha 468 implement user coming back after sell

### PR DESCRIPTION
**Pull request by Danielg1406**
- [x] Creation new table to display the status of the chats and add it to the chips.
- [x] Fix the chips to fit in any device screen.
- [x] Rename ConfirmSale to ConfirmInvitation.
- [x] Refactor ConfirmInvitation to update the user context with the new chat status after confirming.
- [x] Update the user context after sending the invitation to sell a chat to avoid users selling the same chat multiple times.
- [x] Fix the confirm button in my invitations to only appear if there is at least one chat selected.
- [ ] Fix AgreeSale modal to close after the chat is successfully sold.

### CODE REVIEW

**Comment by 552020** [comment]
What we want here basically is to update the context, when the response from send-message and user-to-agreed is received, so that the Chat component, i.e. the different components in it (Chats, Invitations, Summary) show the updated values. Is it right? @Danielg1406

**Comment by Danielg1406** [comment]
> What we want here basically is to update the context, when the response from send-message and user-to-agreed is received, so that the Chat component, i.e. the different components in it (Chats, Invitations, Summary) show the updated values. Is it right? @Danielg1406

correct, the first implementation will move the chats in the invitations section, which all have a pending status, and divide them into the 3 categories (sold, declined, pending) after the user confirms to sell them.

File: src/utils/api/sendMessageHandler.ts [hunk 1]
```diff
@@ -10,7 +10,7 @@ export const sendMessageHandler = async ({
   phoneNumber,
   message,
   backendUrl,
-}: SendMessageHandlerProps): Promise<string[]> => {
+}: SendMessageHandlerProps): Promise<{[key: string]: string[]}> => {
```
Comment (lmangall) [inline src/utils/api/sendMessageHandler.ts:13]: Not sure what is happening here, are you sure this is correct ?

Comment (552020) [reply src/utils/api/sendMessageHandler.ts:13]: image

This is the return type of the mocking function for the send-message endpoint, which should replicate the return type of send-message.

Comment (552020) [reply src/utils/api/sendMessageHandler.ts:13]: As you can see, it is an object with key value pairs, where the key is a string and a value is an array of strings.

Comment (lmangall) [reply src/utils/api/sendMessageHandler.ts:13]: Ok, so at the moment in main we don't handle properly the return of send-message ?

**Review by lmangall — APPROVED** [review]
LGTM

**Pull request merged**